main/modes/quickSettings/quickSettings.c

```diff
@@ -0,0 +1,337 @@
+
+//==============================================================================
+// Strings
+//==============================================================================
+
+static const char quickSettingsName[] = "Settings";
+
+static const char quickSettingsLeds[] = "LEDs ";
+static const char quickSettingsSfx[]  = "Sound Effects ";
+static const char quickSettingsBgm[]  = "Music ";
+
+static const char quickSettingsOn[]    = "On";
+static const char quickSettingsOff[]   = "Off";
+static const char quickSettingsMuted[] = "Muted";
+
+static const char* const quickSettingsOptionsLeds[] = {
+    quickSettingsOff,
+    quickSettingsOn,
+};
+
+static const char* const quickSettingsOptionsAudio[] = {
+    quickSettingsMuted,
+    quickSettingsOn,
+};
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+/// The Swadge mode for Quick Settings
+swadgeMode_t quickSettingsMode = {
+    .modeName                 = quickSettingsName,
+    .wifiMode                 = NO_WIFI,
+    .overrideUsb              = false,
+    .usesAccelerometer        = false,
+    .usesThermometer          = false,
+    .fnEnterMode              = quickSettingsEnterMode,
+    .fnExitMode               = quickSettingsExitMode,
+    .fnMainLoop               = quickSettingsMainLoop,
+    .fnAudioCallback          = NULL,
+    .fnBackgroundDrawCallback = NULL,
+    .fnEspNowRecvCb           = NULL,
+    .fnEspNowSendCb           = NULL,
+    .fnAdvancedUSB            = NULL,
+};
+
+/// All state information for the Quick Settings mode. This whole struct is calloc()'d and free()'d
+/// so that Quick Settings is only using memory while it is active
+quickSettingsMenu_t* quickSettings = NULL;
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+/**
+ * @brief Helper function to easily setup the quick settings options values.
+ *
+ * @param bounds The setting bounds to use, with min for the "off" value
+ * @param currentValue The current setting value, to be used as the "on" value if larger than the min
+ * @param optionValues Pointer to an array of 2 values where the values will be written
+ * @return int32_t Returns the current setting value for convenience
+ */
+static int32_t setupQuickSettingParams(const settingParam_t* bounds, int32_t currentValue, int32_t* optionValues)
+{
+    // Use the minimum and the current value for the toggles
+    // This way, we get the current value back after toggling off.
+    optionValues[0] = bounds->min;
+    optionValues[1] = currentValue;
+
+    // However, if it was already off / at the minimum, we need it to be something, so just use the max
+    if (currentValue <= bounds->min)
+    {
+        optionValues[1] = bounds->max;
+    }
```

Comment:
Yeah, this is where storing the quick setting binary value separately from the actual setting would be nice. Or maybe the quick setting should have the same range of values for each setting as the main settings menu? Binary on/off is quicker, but you get more flexibility with more options, and you can use the current NVS setting as-is.

Reply:
Yeah, I think I'll do it like that. It is a bit weird just being able to toggle it between two values and there's no reason it shouldn't be able to do that.